spec: rename to `image-builder`

supakeen opened the pull request:
`image-builder` will be the main entrypoint into the Image Builder stack. We will rename the repositories later on; but let's start referring to it in its new official name.

supakeen commented:
After this PR merges we likely want to drop 'the old' COPR repository (which is named `-cli`).

mvo5 reviewed (dismissed):
Thank you!

supakeen commented:
I think the RPM builds are failing due to PackIt using the `.packit.yml` from `main`, however some files have been renamed here. @thozza do you know if that's how it works?

If so we can merge this PR and then fix up packit in a separate (followup) PR?

supakeen commented:
> `Makefile:126` needs an adjustment.

Nice catch, done.

schuellerf reviewed (dismissed):
`Makefile:52 / 53` as well as `test/conftest.py` & `HACKING.md` could also be adapted, to remove any `*-cli` traces? :-D

supakeen commented:
> `Makefile:52 / 53` as well as `test/conftest.py` & `HACKING.md` could also be adapted, to remove any `*-cli` traces? :-D

`Makefile:52/53` were explicitly added back in in 87bd9e0d5bf797751febc240847ba7956b0f5054 because it's what the Go macros expect.

I've renamed the test container in `conftest.py`; I've dropped the `-cli` from `HACKING.md`.
